lib/std/heap/memory_pool.zig

```diff
@@ -0,0 +1,183 @@
+const std = @import("../std.zig");
+
+const debug_mode = @import("builtin").mode == .Debug;
+
+pub const MemoryPoolError = error{OutOfMemory};
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Use this instead of a general purpose allocator when you need to allocate a lot of objects
+/// of the same type, as a memory pool outperforms general purpose allocators.
```

Comment:
~~`/// of the same type, as a memory pool outperforms general purpose allocators.`~~
because It outperforms general purpose allocators.